crates/oxc_semantic/src/post_transform_checker.rs

```diff
@@ -163,10 +162,27 @@ struct SemanticData<'s> {
     ids: SemanticIds,
 }
 
+#[derive(Default)]
+struct Errors(Vec<OxcDiagnostic>);
+
+impl Errors {
+    fn push<S: AsRef<str>>(&mut self, message: S) {
+        self.0.push(OxcDiagnostic::error(message.as_ref().trim().to_string()));
+    }
+
+    fn get(self) -> Option<Vec<OxcDiagnostic>> {
```

Comment:
Maybe make this `impl From<Errors> for Option<Vec<OxcDiagnostic>>`?

Reply:
The stack is huge, let's merge it first